app/feature/feature-terminate-insurance/src/main/kotlin/com/hedvig/android/feature/terminateinsurance/step/choose/ChooseInsuranceToTerminateViewModel.kt

```diff
@@ -0,0 +1,183 @@
+import com.hedvig.android.feature.terminateinsurance.data.TerminateInsuranceStep
+import com.hedvig.android.featureflags.FeatureManager
+import com.hedvig.android.featureflags.flags.Feature
+import com.hedvig.android.logger.LogPriority
+import com.hedvig.android.logger.logcat
+import com.hedvig.android.molecule.android.MoleculeViewModel
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flow
+
+internal class ChooseInsuranceToTerminateViewModel(
+  insuranceId: String?,
+  getContractsToTerminateUseCase: GetContractsToTerminateUseCase,
+  terminateInsuranceRepository: TerminateInsuranceRepository,
+  featureManager: FeatureManager,
+) : MoleculeViewModel<ChooseInsuranceToTerminateEvent, ChooseInsuranceToTerminateStepUiState>(
+    initialState = ChooseInsuranceToTerminateStepUiState.Loading,
+    presenter = ChooseInsuranceToTerminatePresenter(
+      insuranceId = insuranceId,
+      getContractsToTerminateUseCase = getContractsToTerminateUseCase,
+      terminateInsuranceRepository = terminateInsuranceRepository,
+      featureManager = featureManager,
+    ),
+  )
+
+private class ChooseInsuranceToTerminatePresenter(
+  private val insuranceId: String?,
+  private val getContractsToTerminateUseCase: GetContractsToTerminateUseCase,
+  private val terminateInsuranceRepository: TerminateInsuranceRepository,
+  private val featureManager: FeatureManager,
+) : MoleculePresenter<ChooseInsuranceToTerminateEvent, ChooseInsuranceToTerminateStepUiState> {
+  @Composable
+  override fun MoleculePresenterScope<ChooseInsuranceToTerminateEvent>.present(
+    lastState: ChooseInsuranceToTerminateStepUiState,
+  ): ChooseInsuranceToTerminateStepUiState {
+    var loadIteration by remember { mutableIntStateOf(0) }
+    var currentSelectedId by remember { mutableStateOf(insuranceId) }
+    var terminationStep: TerminateInsuranceStep? by remember { mutableStateOf(null) }
+    var currentPartialState by remember {
+      mutableStateOf(lastState.toPartialState())
+    }
+
+    CollectEvents { event ->
+      when (event) {
+        ChooseInsuranceToTerminateEvent.RetryLoadData -> {
+          loadIteration++
+        }
+
+        is ChooseInsuranceToTerminateEvent.SelectInsurance -> {
+          currentSelectedId = event.insurance.id
+          if (currentPartialState is PartialUiState.Success) {
+            currentPartialState =
+              (currentPartialState as PartialUiState.Success).copy(selectedInsurance = event.insurance)
+          }
+        }
+      }
+    }
+
+    LaunchedEffect(currentSelectedId) {
+      val id = currentSelectedId
+      if (id != null) {
+        terminationStep = terminateInsuranceRepository.startTerminationFlow(InsuranceId(id)).getOrNull()
+        logcat { "mariia: LaunchedEffect(currentSelectedId) finished, terminationStep: $terminationStep" }
```

Comment:
Same idea here with this log as I write below here, or any logs with your name really, the same applies, keep if you want the log to stay remove the name and give it a title, otherwise if the log was temporary only to debug stuff then feel free to just delete